service/routes.py

```diff
@@ -104,3 +104,34 @@ def list_shopcart_items(customer_id):
     )
 
     return jsonify(shopcart_list), status.HTTP_200_OK
+
+# -----------------------------------------------------------
+# UPDATE PRODUCT QUANTITY
+# -----------------------------------------------------------
+@app.route("/shopcarts/<int:customer_id>/<int:product_id>/<int:quantity>", methods=["PUT"])
+def update_shopcart_item(customer_id, product_id, quantity):
+    """Updates the quantity of an existing product"""
+    app.logger.info(f"Update product-{product_id} in customer-{customer_id}'s cart to {quantity}")
+
+    product_id = int(product_id)
+    customer_id = int(customer_id)
+    quantity = int(quantity)
+
+    shopcart_item = ShopCarts.find_by_customer_id_and_product_id(customer_id, product_id)
+
+    if not shopcart_item:
+        app.logger.error(f"Product-{product_id} doesn't exist in customer-{customer_id}'s cart!")
+        abort(status.HTTP_404_NOT_FOUND, f"Product-{product_id} doesn't exist in the customer-{customer_id}'s cart!")
+
+    shopcart_item.quantities = quantity
+    response_status = status.HTTP_200_OK
+
+    if quantity > 0:
+        shopcart_item.update()
+        app.logger.info(f"Updated Product-{product_id} quantity to {quantity} in customer-{customer_id}'s cart!")
+    else:
+        shopcart_item.delete()
```

Comment:
so we can never receive quantity as 0?

Reply:
When the quantity is 0, we call delete?

Reply:
yea, since there's no reason to have an entry with 0 quantity in the cart. If instead we enforce a positive quantity, then i guess we don't need to call delete.

Reply:
quantity should now be positive. To remove an item delete item api will be called instead of setting it to 0.